pages/Identity/Registry.md

```diff
@@ -0,0 +1,173 @@
+* Handles are subject to homograph attacks
+
+
+## EIP 721
+
+While the NFT standard is not necessary to support, it provides standard ownership transfer and management tools.
+
+TODO: Research the possibilities of EIP 721 support.
+
+## Rejected Alternatives
+
+* Why not the Ethereum Name Service ([ENS](https://ens.domains/))?
+  * Overly complex and expensive for the simple use case of handles
+  * Support can be added later through a custom ENS resolver
+  * Didn't provide a numerical id for efficient graph storage
+  * Still subject to homograph attacks
+
+## Contract Interface
+
+```solidity
+/**
+ * @dev DSNP Registry Interface
+ */
+interface IRegistry {
+    
+    /**
+     * @dev Log when a new handle is registered
+     * @param addr The address the handle is pointing at
+     * @param id The numerical DSNP Id for storing references
+     * @param handle The actual UTF-8 string used for the handle 
+     */
+    event DSNPHandle(address addr, uint64 id, string handle);
+
+    /**
+     * @dev Log when a the handle address is changed
+     * @param addr The new address the handle is pointing at
+     * @param id The numerical DSNP Id for reference 
+     */
+    event DSNPHandleChange(address addr, uint64 id);
+
+    /**
+     * @dev Register a new handle
+     * @param addr Address for the handle to point at
+     * @param handle The handle for the address
+     * 
+     * MUST NOT allow a registration of an addr that is already attached to a different handle
+     * MUST be called by someone who is authorized on the contract via `EIP 1271(addr).isValidSignature`
+     * MUST emit DSNPHandle
+     */
+    function register(address addr, string handle) external;
+
+    /**
+     * @dev Register a new handle By EIP-712 Signature
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param v EIP-155 calculated Signature v value
+     * @param addr Address for the handle to point at
+     * @param handle The handle for the address
+     * 
+     * MUST NOT allow a registration of an addr that is already attached to a different handle
+     * MUST be signed by someone who is authorized on the contract via `EIP 1271(addr).isValidSignature`
+     * MUST emit DSNPHandle
+     */
+    function registerByEIP712Sig(bytes32 r, bytes32 s, uint32 v, address addr, string handle) external;
+
+    /**
+     * @dev Alter a handle resolution
+     * @param newAddr New address for the handle to point at
+     * @param handle The handle to alter
+     * 
+     * MUST NOT allow a registration of an newAddr that is already attached to a different handle
+     * MUST be called by someone who is authorized on the contract via `EIP 1271(oldAddr).isValidSignature`
+     * TODO: FIX THE ISSUE OF newAddr not being a part of the creation
+     * MUST emit DSNPHandleOwnershipChange
+     */
+    function setAddr(address newAddr, string handle) external;
```

Comment:
IF the handle is merely for discovery, should there be a way to change your handle?

Should handles be separate (more like an NFT)?

Reply:
Added a way to change handles.